tools/make_partition

```diff
@@ -0,0 +1,214 @@
+            col = line.split()[8]
+            dst_port = col.split(':')[-1]
+            if dst_port in port_to_pid:
+                src_port = col.split('->')[0].split(':')[-1]
+                port_to_pid[src_port] = line.split()[1]
+
+    pids_A, pids_B = [], []
+    for port, pid in port_to_pid.items():
+        if port in A:
+            pids_A.append(pid)
+        elif port in B:
+            pids_B.append(pid)
+
+    port_to_owner = {}
+    for port, pid in port_to_pid.items():
+        if pid in pids_A:
+            port_to_owner[port] = 'A'
+        elif pid in pids_B:
+            port_to_owner[port] = 'B'
+        else:
+            # If there is an owner we don't understand, just skip it.
+            # print("Unknown owner for %s" % pid)
+            pass
+
+    return port_to_owner
+
+
+def _block_tuples(lsof, port_to_owner):
+    """Returns IP and a list of (src_port, dst_port)-tuples."""
+    ret = []
+    for line in [l for l in lsof if '(ESTABLISHED)' in l]:
+        col = line.split()[8]
+        src = col.split('->')[0].split(':')[-1]
+        dst = col.split(':')[-1]
+        if port_to_owner.get(src) and port_to_owner.get(dst):
+            ip = col.split('->')[0].split(':')[0]
+            if port_to_owner[src] != port_to_owner[dst]:
+                if (dst, src) not in ret:
+                    ret.append((src, dst))
+    return ip, ret
+
+
+def _block_preamble_darwin(ip, ports):
+    """Block new connections to listening ports on Darwin."""
+    tpl = "block return in proto tcp from %s port %s flags S/S"
+    return [tpl % (ip, port) for port in ports]
+
+
+def _block_preamble_linux(ip, ports):
+    """Block new connections to listening ports on Linux."""
+    tpl = ("-A INPUT -p tcp --syn -m state --state NEW "
+           "-d %s --dport %s -j REJECT --reject-with tcp-reset")
+    return ["*filter"] + [tpl % (ip, port) for port in ports]
+
+
+def _block_pair_darwin(ip, blocks):
+    """Block a known connection from A to B on Darwin."""
+    tpl = "block return in proto tcp from %s port %s to %s port %s"
+    ret = []
+    for src, dst in blocks:
+        ret.append(tpl % (ip, src, ip, dst))
+        ret.append(tpl % (ip, dst, ip, src))
+    return ret
+
+
+def _block_pair_linux(ip, blocks):
+    """Block a known connection from A to B on Linux."""
+    tpl = ("-A INPUT -p tcp -s %s --sport %s -d %s "
+           "--dport %s -j REJECT --reject-with tcp-reset")
+    ret = []
+    for src, dst in blocks:
+        ret.append(tpl % (ip, src, ip, dst))
+        ret.append(tpl % (ip, dst, ip, src))
+    return ret
+
+
+def _block_appendix_linux():
+    """iptables-save requires "COMMIT" at the end."""
+    return ["COMMIT"]
+
+
+def main():
+    parser = argparse.ArgumentParser(
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description='Generate firewall rules to create a partition.',
+            epilog=__doc__)
+    parser.add_argument('A', help='Ports of A, e.g. 3000,3001-3002')
```

Comment:
Yes, a range. Worked for me:

```sh
sudo lsof -Pnni | ./tools/make_partition 3000-3003,3008 3004-3007 | sudo pfctl -emf -
```

And output from ringpop-admin partitions:

```
 11:28:31.138   Checksum     # Nodes   # Alive   # Suspect   # Faulty   Sample Host
                91644736     5         5         0           4          127.0.0.1:3000
                1106031481   4         4         0           5          127.0.0.1:3004```


Can you clarify what didn't work?